### ISSUE TITLE
Adding undetected screen resolution is broken in nixos-unstable

### ISSUE THREAD

**Issue by magnetophon** [NixOS/nixpkgs#27168]
## Issue description

I have some hardware where the correct screen resolution is not detected.
I've been able to work around that by doing:
```
xrandr --newmode "1920x1080_60.00"  173.00  1920 2048 2248 2576  1080 1083 1088 1120 -hsync +vsync
xrandr --addmode VGA-0 1920x1080_60.00
xrandr --output VGA-0 --mode 1920x1080_60.00
```
After an upgrade to nixos-unstable, that leads to a screen where the image is too high up, and cut off on the right.
IOW: the bottom and right side of my screen are black.

Can anyone help me find out if this is a NixOS bug, or an upstream bug?
Any hints for further troubleshooting are also appreciated.

@bkchr @IreneKnapp @eqyiel 
A quick github search suggests you also have hardware that does not detect the right resolution.
Are any of you on nixos-unstable?
Are you having issues?

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
17.09pre110213.01c3847b9c (Hummingbird)
* Nix version: (run `nix-env --version`)
nix-env (Nix) 1.11.11
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
"17.09pre110213.01c3847b9c"

* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`)
build-use-sandbox = true

**Comment by bkchr** [comment]
Do you use the kernelmodesetting driver? I think that this is an upstream bug in the kernel. However, I switched to the preferred way and scale up everything and use the native resolution of my monitor.

However, I also still have the following script, maybe it helps you:
```
modeline=$(cvt -v 1920 1080 | grep Modeline | sed -e "s/Modeline //")
xrandr --newmode $modeline
modename=$(echo $modeline | awk '{ print $1; }')
xrandr --addmode eDP-1 $modename
```
But it should do the same as your old workaround :/

**Comment by magnetophon** [comment]
@bkchr Thanks.  
IDK if I'm using that driver. How do I find out?
AFAIK this is my monitors native resolution.

**Comment by bkchr** [comment]
Ahh okay, for me my system always detected the native resolution.

To check if you use modesetting you could try the following: https://carlosbecker.com/posts/how-to-enable-kms-on-i915-graphiccard-archlinux/
But that will only work if your GPU is a Intel GPU. Do you have a Intel GPU?

**Comment by magnetophon** [comment]
It looks like I am using KMS, if I'm not mistaken:
``01:05.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] RS880 [Radeon HD 4200]``
```
lsmod | grep radeon
radeon               1466368  3
ttm                    86016  1 radeon
drm_kms_helper        122880  1 radeon
drm                   294912  6 radeon,ttm,drm_kms_helper
i2c_algo_bit           16384  1 radeon
i2c_core               53248  5 i2c_piix4,i2c_algo_bit,radeon,drm_kms_helper,drm
```

**Comment by magnetophon** [comment]
Seems like I should just try with ``nomodeset`` in grub, right?

**Comment by bkchr** [comment]
Yeah you could try this, however, if possible I would stick to modesetting. 

You could try the following:
Use amdgpu.audio=0 as kernel parameter (Source: https://wiki.archlinux.org/index.php/AMDGPU)

**Comment by magnetophon** [comment]
thanks, I'll try those then.

**Comment by magnetophon** [comment]
I tried both ``amdgpu.audio=0`` and ``nomodeset``.
Same result.

**Comment by bkchr** [comment]
This all really sounds like an upstream bug. Maybe you can report your bug [here?](https://bugs.freedesktop.org/buglist.cgi?quicksearch=amdgpu)
But I really don't know where to report these bugs^^ Or maybe some mailinglist. 
Ahh, one idea maybe, do you tried to update your kernel? I assume that you still use 4.9?

**Comment by magnetophon** [comment]
@bkchr Thanks for all your help.

I'm using ``Linux mixos 4.9.33-rt23 #1-NixOS SMP PREEMPT RT Sat Jun 17 04:43:47 UTC 2017 x86_64 GNU/Linux``.
It's a real time kernel, Nixified by (musnix)[https://github.com/musnix/musnix].
It's the latest one available there, but I could try downgrading.

**Comment by bkchr** [comment]
Try downgrading, but I would not recommend, because amdgpu is relative new and every new release brings new features and bug fixes. I would really recommend an upgrade, maybe just try if the bug goes away and then you need wait for a new release of your real time kernel ^^

**Comment by magnetophon** [comment]
I'm already building 4.8 rt.
If that doesn't work I'll try a newer non-rt one, to see if it's indeed kernel related.

**Comment by magnetophon** [comment]
4.8 did not fix it, but 4.9.35 non rt did.
I guess that makes it not a nixpkgs issue, so I'll close it.

I'm currently building 4.9.35 rt....
